File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

int main(void)
{
    string t = get_string("Text: ");
    int l = 0, w = 1, s = 1;
    for (int i = 0; i < strlen(t); i++)
    {
        int letter = isalpha(t[i]) / 1024;
        int word = t[i] == ' ';
        int sentence = (t[i] == '.' || t[i] == '?' || t[i] == '!') && (t[i + 1] == ' ');
        l += letter;
        w += word;
        s += sentence;
    }
    float fl = (float) l, fw = (float) w, fs = (float) s;
    float idx = 0.0588 * (fl / fw * 100.0) - 0.296 * (fs / fw * 100.0) - 15.8;
    int grade = round(idx);
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %d\n", grade);
    }
}

File: pset6/readability/readability.py
letters = words = sentences = 0
for c in input('Text: '):
    if c in '.!?':
        sentences += 1
    elif c == ' ':
        words += 1
    elif c.isalpha():
        letters += 1
words += 1
grade = round((5.88 * letters - 29.6 * sentences) / words - 15.8)
print(f'Before Grade 1' if grade < 1 else 'Grade 16+' if grade >= 16 else f'Grade {grade}')
